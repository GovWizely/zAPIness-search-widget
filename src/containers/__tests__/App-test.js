import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import ConnectedApp, { App } from '../App';

const { Map } = require('immutable');

describe('containers/App', () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const filters = Map({
    categories: [],
    filters: []
  });
  const toggle = Map({
    key: undefined,
    show: false
  });

  const initialState = Map({
    categories: [],
    keyword: '',
    offset: 0,
    pageNum: 1,
    filters,
    data: undefined,
    error: [],
    query: Map({
      keyword: '',
      offset: 0,
      filters: []
    }),
    isFetching: false,
    toggle
  });

  const store = mockStore(initialState);

  const handleSelect = jest.fn();
  const getCategories = jest.fn();
  const updateFields = jest.fn();
  const toggleResultHandler = jest.fn();
  const previewResult = jest.fn();

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App
          query={initialState}
          toggle={toggle}
          handleSelect={handleSelect}
          getCategories={getCategories}
          updateFields={updateFields}
          toggleResultHandler={toggleResultHandler}
          previewResult={previewResult}
          isFetching={false}
          filters={filters}
        />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatch actions to store', () => {
    shallow(
      <Provider store={store}>
        <ConnectedApp />
      </Provider>
    );

    expect(store.getActions().length).toEqual(1);
  });

  it('renders no result during data fetching', () => {
    const app = shallow(
      <App
        query={initialState}
        toggle={toggle}
        handleSelect={handleSelect}
        getCategories={getCategories}
        updateFields={updateFields}
        toggleResultHandler={toggleResultHandler}
        previewResult={previewResult}
        isFetching
        filters={filters}
      />
    );

    expect(app.find('div.__sw-result__').exists()).toBe(false);
  });
});
