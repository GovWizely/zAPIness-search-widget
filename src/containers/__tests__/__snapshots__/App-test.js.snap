// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/App renders correctly 1`] = `
<div
  className="__sw-container__"
  style={
    Object {
      "padding": "20px",
      "width": "100%",
    }
  }
>
  <div
    className="container"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <form
      className="__sw-input__"
      onSubmit={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "height": "46px",
            "position": "relative",
            "verticalAlign": "top",
            "width": "80%",
          }
        }
      >
        <input
          className="__input__"
          name="keyword"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Search for keyword..."
          style={
            Object {
              "borderColor": "#f9f9f9",
              "borderRadius": 3,
              "borderStyle": "solid",
              "borderWidth": 1,
              "boxSizing": "border-box",
              "display": "inline-block",
              "fontFamily": "\\"Helvetica Neue\\",Helvetica,Arial,sans-serif",
              "fontSize": "15px",
              "fontWeight": 300,
              "lineHeight": "28px",
              "padding": "8px 10px",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <button
        className="__sw-advanced-search__"
        data-radium={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "alignItems": "center",
            "background": "#f9f9f9",
            "border": "none",
            "borderRadius": 3,
            "color": "#7c7c7c",
            "display": "inline-block",
            "flex": 1,
            "height": "46px",
            "justifyContent": "center",
            "marginLeft": "10px",
            "overflow": "hidden",
            "padding": "0 5px",
            "width": "46px",
          }
        }
        type="button"
      >
        <span
          data-radium={true}
        >
          <img
            alt="Add"
            data-radium={true}
            src="test-file-stub"
            style={
              Object {
                "color": "#BDBDBD",
                "height": "20px",
                "width": "20px",
              }
            }
          />
        </span>
      </button>
    </form>
  </div>
</div>
`;
